android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -1693,10 +1810,17 @@ public void callInjectedJavaScriptBeforeContentLoaded() {
 
     public void onMessage(String message) {
       ReactContext reactContext = (ReactContext) this.getContext();
-      RNCWebView mContext = this;
-
-      if (mRNCWebViewClient != null) {
-        WebView webView = this;
+      InternalWebView mContext = this;
+      WebView webView = this;
+
+      if (webViewKey != null && mRNCWebViewClient != null) {
+        reactContext.runOnUiQueueThread(() -> {
+          WritableMap data = mRNCWebViewClient.createWebViewEvent(webView, webView.getUrl());
+          data.putString("webViewKey", webViewKey);
+          data.putString("data", message);
+          reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit("ReactNativeWebViewOnMessageWithWebViewKey", data);
+        });
+      } else if (mRNCWebViewClient != null) {
```

Comment:
nit: It looks like `onMessage` is basically a no-op if `mRNCWebViewClient` is `null`...? So perhaps instead of checking it in both the `if` and `else if` branches, there could be a higher-level check (e.g., just return if it's `null`)